plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/manifest/ManifestManager.kt

```diff
@@ -0,0 +1,128 @@
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.intellij.openapi.util.SystemInfo
+import com.intellij.util.io.HttpRequests
+import com.intellij.util.system.CpuArch
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.warn
+import java.net.HttpURLConnection
+import java.net.URL
+
+class ManifestManager {
+    private val cloudFrontUrl = "https://aws-toolkit-language-servers.amazonaws.com/q-context/manifest.json"
+    val currentVersion = "0.1.2"
+    val currentOs = getOs()
+    private val arch = CpuArch.CURRENT
+    private val mapper = jacksonObjectMapper()
+
+    data class TargetContent(
+        @JsonIgnoreProperties(ignoreUnknown = true)
+        @JsonProperty("filename")
+        val filename: String? = null,
+        @JsonProperty("url")
+        val url: String? = null,
+        @JsonProperty("hashes")
+        val hashes: List<String>? = emptyList(),
+        @JsonProperty("bytes")
+        val bytes: Number? = null
+    )
+
+    data class VersionTarget(
+        @JsonIgnoreProperties(ignoreUnknown = true)
+        @JsonProperty("platform")
+        val platform: String? = null,
+        @JsonProperty("arch")
+        val arch: String? = null,
+        @JsonProperty("contents")
+        val contents: List<TargetContent>? = emptyList()
+    )
+    data class Version(
+        @JsonIgnoreProperties(ignoreUnknown = true)
+        @JsonProperty("serverVersion")
+        val serverVersion: String? = null,
+        @JsonProperty("isDelisted")
+        val isDelisted: Boolean? = null,
+        @JsonProperty("targets")
+        val targets: List<VersionTarget>? = emptyList()
+    )
+
+    data class Manifest(
+        @JsonIgnoreProperties(ignoreUnknown = true)
+        @JsonProperty("manifestSchemaVersion")
+        val manifestSchemaVersion: String? = null,
+        @JsonProperty("artifactId")
+        val artifactId: String? = null,
+        @JsonProperty("artifactDescription")
+        val artifactDescription: String? = null,
+        @JsonProperty("isManifestDeprecated")
+        val isManifestDeprecated: Boolean? = null,
+        @JsonProperty("versions")
+        val versions: List<Version>? = emptyList()
+    )
+
+    fun getManifest(): Manifest? {
+        val url = URL(cloudFrontUrl)
+        val connection = url.openConnection() as HttpURLConnection
+        connection.requestMethod = "GET"
+        connection.setRequestProperty("Accept", "application/json")
+        connection.setRequestProperty("Content-Type", "application/json")
+        return fetchFromRemoteAndSave()
+    }
+
+    private fun readManifestFile(content: String): Manifest? {
+        try {
+            return mapper.readValue<Manifest>(content)
+        } catch (e: Exception) {
+            logger.warn { "error parsing manifest file for project context ${e.message}" }
+            return null
+        }
```

Comment:
fyi we could use util func
```
logger.tryOrNull("error while parsing manifest file for.....", level = Level.WARN) { mapper.readValue(content) }
```

Reply:
```suggestion
      return   logger.tryOrNull("error while parsing manifest file for.....", level = Level.WARN) { mapper.readValue(content) }
```

Reply:
so it could simplify to 
```
private fun readManifestFile(content: String): Manifest? = logger.tryOrNull("error while parsing manifest file for.....", level = Level.WARN) { mapper.readValue(content) }
```

Reply:
not critical